# AirBnB Data Analysis Project

This is a capstone project for the Code:You Data Analysis course. This project 
will analyze Airbnb data to uncover useful correlations/trends and guide 
decision making in investing in STR properties.

## Project Goal

**Determine the optimal location and house size for a short term rental investment 
property in Seattle.**

The end result of this project is to classify properties in terms of STR 
Performance vs. Purchase Price:
- High STR Performance, Low Purchase Price
- High STR Performance, High Purchase Price
- Low STR Performance, Low Purchase Price
- Low STR Perormance, High Purchase Price

This will allow us to narrow our search for an investment proprety to the 
neighborhoods and house size that are most likely to be profitable.

In this data discovery notebook, I will:
1. Research key metrics for Short Term Rentals.
1. Research the Seattle housing market.
1. Determine data needed to support.
1. Review available data.
1. Identify the type of analysis that can be done and list the questions that 
can be answered.
1. List the cleaning steps that will be needed.


## 1. Short Term Rental Metrics Research

A quick google search gives us some of the key metrics used to evaluate the 
financial performance of a short term rental property. NOI and RPP appear to be
most related to our goal of finding a good investment property.

> **Net operating income (NOI)** is another key Airbnb metric for Airbnb hosts to evaluate the profitability of their listing. NOI measures the total revenue generated by a property from all sources and subtracts operating expenses, including management, maintenance, and cleaning fees. Hosts can use NOI to accurately determine their net profit and make informed decisions about pricing strategy, cost-saving measures, or investing in property improvements. By focusing on maximizing their NOI, hosts can drive profitability on the Airbnb platform and make sure they achieve sustainable success. Therefore, tracking NOI is an essential factor for any host looking to optimize their financial performance and grow their vacation rental business.

In order to calculate Net Operating Income, we will need to determine the monthly revenue, monthly mortgage payment, and the property management fee. We can use estimates for the mortgage rate and property management. We can calculte the monthly revenue using the nightly rate and the occupancy %.

NOI = (Nightly Rate * Days Occupied) - (Management Fee % * (Nightly Rate * Days Occupied) - Mortgage

> **Revenue per property (RPP)** calculates the total earnings generated by a property within a defined timeframe. Monitoring RPP allows hosts to discern which listings are the highest revenue generators, enabling them to fine-tune pricing and marketing strategies accordingly. Furthermore, RPP aids hosts in assessing the financial feasibility of investing in new properties or enhancing existing ones. By optimizing RPP, hosts can enhance operational efficiency on Airbnb and thrive in the fiercely competitive vacation rental market. Tracking RPP becomes a pivotal aspect for hosts seeking to maximize financial performance and expand their business endeavors.

We can calculate the Revenue per Property by using the nightly rate and the occupancy.

Revenue Per Property = Nightly Rate * # Days Occupied

Sources: Short Term Rental KPIs

- [Key STR Metrics](https://hostify.com/blog/key-airbnb-performance-metrics-hosts-should-track)



## 2. Seattle Real Estate Market Research 

In order to determine the best location and size of house to buy for a short
term rental, we will need a way to compare the relative price of different
properties. A quick google search yielded house prices for the Seattle Market 
broken down by neighborhood and number of bedrooms.


### House Sale Price by Neighborhood


In [1]:
import pandas as pd

cost_by_neighborhood = pd.read_csv("../data/raw/Home_Cost_By_Neighborhood.csv")
cost_by_neighborhood

Map Area New Listings, Year-to-Date  \
0                       Auburn                        704   
1            Ballard/Greenlake                      2,125   
2         Bellevue–East of 405                        779   
3         Bellevue–West of 405                        460   
4            Belltown/Downtown                        561   
5   Black Diamond/Maple Valley                      1,003   
6         Burien/Normandy Park                        854   
7              Central Seattle                      1,492   
8       Dash Point/Federal Way                        687   
9           Des Moines/Redondo                        442   
10      East of Lake Sammamish                      1,640   
11     East Side–South of I-90                        833   
12                    Enumclaw                        270   
13            Jovita/West Hill                        336   
14         Juanita/Woodinville                      1,464   
15                        Kent                      1,087   
16      Kirkland/Bridle Trails                        836   
17            Lake Forest Park                        363   
18               Mercer Island                        257   
19               North Seattle                      1,080   
20         Queen Anne/Magnolia                      1,063   
21           Redmond/Carnation                        615   
22          Renton-Benson Hill                        648   
23            Renton-Highlands                        646   
24    Richmond Beach/Shoreline                        375   
25                 Skyway Area                        240   
26            SODO/Beacon Hill                        437   
27           Southeast Seattle                        507   
28               Vashon Island                         77   
29                West Seattle                      1,502   

   Closed Sales, Year-to-Date Median Price, Year-to-Date  \
0                         569                   $500,000   
1                       1,635                   $819,000   
2                         575                 $1,035,000   
3                         357                 $1,610,000   
4                         366                   $635,000   
5                         788                   $685,250   
6                         721                   $549,000   
7                       1,018                   $810,000   
8                         570                   $500,000   
9                         395                   $510,000   
10                      1,310                 $1,088,000   
11                        638                 $1,150,000   
12                        214                   $525,000   
13                        275                   $578,604   
14                      1,169                   $900,000   
15                        859                   $560,000   
16                        662                 $1,347,000   
17                        310                   $758,000   
18                        191                 $1,865,000   
19                        846                   $865,500   
20                        818                   $921,250   
21                        459                   $990,000   
22                        553                   $600,000   
23                        508                   $739,500   
24                        272                   $735,750   
25                        170                   $579,995   
26                        325                   $669,950   
27                        372                   $720,000   
28                         61                   $760,000   
29                      1,199                   $690,000   

   Average Price, Year-to-Date Closed Sales, 2020 Median Price, 2020  \
0                     $525,549              1,061           $444,000   
1                     $878,551              2,841           $760,000   
2                   $1,136,090              1,073    

### House Sale Price by Number of Bedrooms


In [2]:
import pandas as pd

cost_by_bedrooms = pd.read_csv("../data/raw/Home_Cost_by_Bedrooms.csv")
cost_by_bedrooms

Bedrooms  Average Size (sf) Median Sale Price Price Difference  \
0          1                NaN          $678,667        -$221,333   
1          2             1321.0          $900,000        -$212,000   
2          3             1906.0        $1,112,000         $212,000   
3          4             2420.0        $1,402,500         $290,000   
4  5 or more             2930.0        $1,565,000         $162,000   

   Price / sf  Number of sales  
0         NaN              NaN  
1       681.0             77.0  
2       583.0            193.0  
3       579.0            110.0  
4       534.0             51.0

**Sources** 

Seattle Real Estate Market Data

- [Home Price by Neighborhood](https://www.seattlemet.com/home-and-real-estate/how-much-do-homes-cost-in-seattle-area-neighborhoods-real-estate)
- [Home Price by Number of Bedrooms](https://www.myseattlehomesearch.com/blog/seattle-house-prices-versus-the-number-of-bedrooms/#:~:text=Not%20shown%20in%20the%20data,just%2012%25%20of%20the%20market.)



## 3. Data Needed To Support Topic

Housing Market
- Sales Price 
- number of bedrooms
- square footage
- location

AirBnB Data
- number / percent of booked days 
- number of bedrooms
- square footage
- location
- nightly rate

Other (Needed for NOI calculations)
- Mortgage rate - we can use 7.5% and adjust if needed
- STR management fee - we can use 10%

## 4. Data Discovery - Kaggle AirBnB Data

The next step is to see what data is available in the Kaggle data set.

Sources:
- [Kaggle Seattle Listing Data Set](https://www.kaggle.com/datasets/airbnb/seattle)


There are three primary data sets available from the Kaggle data.
- AirBnB Listings
- AirBnB Reviews
- AirBnB Calendar

### Listings

The listings data set contains AirBnB listings from Seattle, Washington. To 
start off the analysis we will load and preview the listings data.

In [3]:
import pandas as pd

listings = pd.read_excel("../data/raw/Tableau Full Project.xlsx", sheet_name=0)
listings

id                            listing_url       scrape_id  \
0       241032    https://www.airbnb.com/rooms/241032  20160104002432   
1       953595    https://www.airbnb.com/rooms/953595  20160104002432   
2      3308979   https://www.airbnb.com/rooms/3308979  20160104002432   
3      7421966   https://www.airbnb.com/rooms/7421966  20160104002432   
4       278830    https://www.airbnb.com/rooms/278830  20160104002432   
...        ...                                    ...             ...   
3813   8101950   https://www.airbnb.com/rooms/8101950  20160104002432   
3814   8902327   https://www.airbnb.com/rooms/8902327  20160104002432   
3815  10267360  https://www.airbnb.com/rooms/10267360  20160104002432   
3816   9604740   https://www.airbnb.com/rooms/9604740  20160104002432   
3817  10208623  https://www.airbnb.com/rooms/10208623  20160104002432   

     last_scraped                                 name  \
0      2016-01-04         Stylish Queen Anne Apartment   
1      2016-01-04   Bright & Airy Queen Anne Apartment   
2      2016-01-04  New Modern House-Amazing water view   
3      2016-01-04                   Queen Anne Chateau   
4      2016-01-04       Charming craftsman 3 bdm house   
...           ...                                  ...   
3813   2016-01-04   3BR Mountain View House in Seattle   
3814   2016-01-04    Portage Bay View!-One Bedroom Apt   
3815   2016-01-04    Private apartment view of Lake WA   
3816   2016-01-04    Amazing View with Modern Comfort!   
3817   2016-01-04            Large Lakefront Apartment   

                                                summary  \
0                                                   NaN   
1     Chemically sensitive? We've removed the irrita...   
2     New modern house built in 2013.  Spectacular s...   
3     A charming apartment that sits atop Queen Anne...   
4     Cozy family craftman house in beautiful neighb...   
...                                                 ...   
3813  Our 3BR/2BA house boasts incredible views of t...   
3814  800 square foot 1 bedroom basement apartment w...   
3815  Very comfortable lower unit. Quiet, charming m...   
3816  Cozy studio condo in the heart on Madison Park...   
3817  All hardwood floors, fireplace, 65" TV with Xb...   

                                                  space  \
0     Make your self at home in this charming one-be...   
1     Beautiful, hypoallergenic apartment in an extr...   
2     Our house is modern, light and fresh with a wa...   
3                                                   NaN   
4     Cozy family craftman house in beautiful neighb...   
...                                                 ...   
3813  Our 3BR/2BA house bright, stylish, and wheelch...   
3814  This space has a great view of Portage Bay wit...   
3815                                                NaN   
3816  Fully furnished unit to accommodate most needs...   
3817                                                NaN   

                                            description experiences_offered  \
0     Make your self at home in this charming one-be...                none   
1     Chemically sensitive? We've removed the irrita...                none   
2     New modern house built in 2013.  Spectacular s...                none   
3     A charming apartment that sits atop Queen Anne...                none   
4     Cozy family craftman house in beautiful neighb...                none   
...                                                 ...                 ...   
3813  Our 3BR/2BA house boasts incredible views of t...                none   
3814  800 square foot 1 bedroom basement apartment w...                none   
3815  Very comfortable lower unit. Quiet, charming m...                none   
3816  Cozy studio condo in the heart on Madison Park...                none   
3817  All hardwood floors, fireplace, 65" TV with Xb...                none   

                                  neighborhood_overview  ...  \
0      

In [4]:
listings.shape

(3818, 92)

There are 3,818 AirBnB listings included in the data. Each listing has 92 attributes.

Next we will look at the list of attributes available.

In [5]:
listings.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3818 entries, 0 to 3817
Data columns (total 92 columns):
 #   Column                            Non-Null Count  Dtype         
---  ------                            --------------  -----         
 0   id                                3818 non-null   int64         
 1   listing_url                       3818 non-null   object        
 2   scrape_id                         3818 non-null   int64         
 3   last_scraped                      3818 non-null   datetime64[ns]
 4   name                              3818 non-null   object        
 5   summary                           3639 non-null   object        
 6   space                             3249 non-null   object        
 7   description                       3818 non-null   object        
 8   experiences_offered               3818 non-null   object        
 9   neighborhood_overview             2786 non-null   object        
 10  notes                             2206 non-null 

**Numeric Fields**

In [6]:
listings.describe()

id     scrape_id                   last_scraped  \
count  3.818000e+03  3.818000e+03                           3818   
mean   5.550111e+06  2.016010e+13  2016-01-04 00:00:00.000000256   
min    3.335000e+03  2.016010e+13            2016-01-04 00:00:00   
25%    3.258256e+06  2.016010e+13            2016-01-04 00:00:00   
50%    6.118244e+06  2.016010e+13            2016-01-04 00:00:00   
75%    8.035127e+06  2.016010e+13            2016-01-04 00:00:00   
max    1.034016e+07  2.016010e+13            2016-01-04 00:00:00   
std    2.962660e+06  0.000000e+00                            NaN   

            host_id                     host_since  host_response_rate  \
count  3.818000e+03                           3816         3295.000000   
mean   1.578556e+07  2013-09-18 17:26:02.264151040            0.948868   
min    4.193000e+03            2008-11-10 00:00:00            0.170000   
25%    3.275204e+06            2012-08-17 00:00:00            0.980000   
50%    1.055814e+07            2013-12-12 12:00:00            1.000000   
75%    2.590309e+07            2015-01-14 00:00:00            1.000000   
max    5.320861e+07            2016-01-03 00:00:00            1.000000   
std    1.458382e+07                            NaN            0.118667   

       host_acceptance_rate  host_listings_count  host_total_listings_count  \
count           3045.000000          3816.000000                3816.000000   
mean               0.999672             7.157757                   7.157757   
min                0.000000             1.000000                   1.000000   
25%                1.000000             1.000000                   1.000000   
50%                1.000000             1.000000                   1.000000   
75%                1.000000             3.000000                   3.000000   
max                1.000000           502.000000                 502.000000   
std                0.018122            28.628149                  28.628149   

          latitude  ...  review_scores_rating  review_scores_accuracy  \
count  3818.000000  ...           3171.000000             3160.000000   
mean     47.628961  ...             94.539262                9.636392   
min      47.505088  ...             20.000000                2.000000   
25%      47.609418  ...             93.000000                9.000000   
50%      47.623601  ...             96.000000               10.000000   
75%      47.662694  ...             99.000000               10.000000   
max      47.733358  ...            100.000000               10.000000   
std       0.043052  ...              6.606083                0.698031   

       review_scores_cleanliness  review_scores_checkin  \
count                3165.000000            3160.000000   
mean                    9.556398               9.786709   
min                     3.000000               2.000000   
25%                     9.000000              10.000000   
50%                    10.000000              10.000000   
75%                    10.000000              10.000000   
max                    10.000000              10.000000   
std                     0.797274               0.595499   

       review_scores_communication  review_scores_location  \
count                  3167.000000             3163.000000   
mean                      9.809599                9.608916   
min                       2.000000                4.000000   
25%                      10.000000                9.000000   
50%                      10.000000               10.000000   
75%                      10.000000               10.000000   
max                      10.000000               10.000000   
std                       0.568211                0.629053   

       review_scores_value  license  calculated_host_listings_count  \
count          3162.000000      0.0                     3818.000000   
mean              9.452245      NaN                        2.946307   
min               2.000000      NaN                        1.00

In [7]:
listings.price.describe()

count    3818.000000
mean      127.976166
std        90.250022
min        20.000000
25%        75.000000
50%       100.000000
75%       150.000000
max      1000.000000
Name: price, dtype: float64

All of the listings have a price. Prices range from $20 - $1000 with the average being $127.98.

In [8]:
listings.bedrooms.describe()

count    3812.000000
mean        1.307712
std         0.883395
min         0.000000
25%         1.000000
50%         1.000000
75%         2.000000
max         7.000000
Name: bedrooms, dtype: float64

It looks like all but 6 of the listings records have data for the number of bedrooms. 

In [9]:
listings.square_feet.describe()

count      97.000000
mean      854.618557
std       671.404893
min         0.000000
25%       420.000000
50%       750.000000
75%      1200.000000
max      3000.000000
Name: square_feet, dtype: float64

Only 97 records have square footage data. Based on this it looks like we will have to use the number of bedrooms as a proxy for house size.

In [16]:
listings.availability_30.describe()

count    3818.000000
mean       16.786276
std        12.173637
min         0.000000
25%         2.000000
50%        20.000000
75%        30.000000
max        30.000000
Name: availability_30, dtype: float64

All of the listings have availability data. If we assume that this is just tracking nights booked and does not include maintenance or other non-booked nights, we should be able ot use this to get the occupancy %.

**Categorical Fields**

In [10]:
listings.neighbourhood_group_cleansed.unique()

array(['Queen Anne', 'Ballard', 'Other neighborhoods', 'Cascade',
       'Central Area', 'University District', 'Downtown', 'Magnolia',
       'West Seattle', 'Interbay', 'Beacon Hill', 'Rainier Valley',
       'Delridge', 'Seward Park', 'Northgate', 'Capitol Hill',
       'Lake City'], dtype=object)

In [11]:
listings.neighbourhood_group_cleansed.value_counts()

neighbourhood_group_cleansed
Other neighborhoods    794
Capitol Hill           567
Downtown               530
Central Area           369
Queen Anne             295
Ballard                230
West Seattle           203
Rainier Valley         159
University District    122
Beacon Hill            118
Cascade                 89
Northgate               80
Delridge                79
Lake City               67
Magnolia                61
Seward Park             44
Interbay                11
Name: count, dtype: int64

Since both the AirBnB data and the real estate market data have neighborhood, we can use this as a proxy for location.

In [12]:
listings.property_type.unique()

array(['Apartment', 'House', 'Cabin', 'Condominium', 'Camper/RV',
       'Bungalow', 'Townhouse', 'Loft', 'Boat', 'Bed & Breakfast',
       'Other', 'Dorm', 'Treehouse', 'Yurt', 'Chalet', 'Tent', nan],
      dtype=object)

In [13]:
listings.property_type.value_counts()

property_type
House              1733
Apartment          1708
Townhouse           118
Condominium          91
Loft                 40
Bed & Breakfast      37
Other                22
Cabin                21
Camper/RV            13
Bungalow             13
Boat                  8
Tent                  5
Treehouse             3
Dorm                  2
Chalet                2
Yurt                  1
Name: count, dtype: int64

In [14]:
listings.room_type.unique()

array(['Entire home/apt', 'Private room', 'Shared room'], dtype=object)

We will want to use room_type to filter out anything that isn't an 'Entire home/apt'.

In [15]:
listings.room_type.value_counts()

room_type
Entire home/apt    2541
Private room       1160
Shared room         117
Name: count, dtype: int64

It looks like the AirBnB data is going to have the fields needed to answer the 
investment questions.

## 5. Questions that can be answered

The purpose of this project is to analyze AirBnB data to determine the most 
attractive attributes of a short term rental investment.


- Is an Air BnB in seattle a good investment?
- Are there any cabins in a specific neighborhood?
- How much profit can we get from an investment?
- Do properties with lower fees get higher reviews?
- Which locations have the most available purchases compared to common travel spots?
- Where is the cheapest AirBnB near the Seattle Seahawks stadium?
- What kind of properties get booked the most?

## 6. Cleaning Needed

### Remove Columns

**AirBnB Listings Fields to Keep and Clean**

- id
- name
- street
- neighbourhood 
- neighbourhood_cleansed 
- neighbourhood_group_cleansed 
- property_type
- room_type
- accommodates
- bathrooms
- bedrooms
- beds  
- bed_type
- square_feet
- price
- weekly_price
- monthly_price
- calendar_updated
- has_availability
- availability_30
- availability_60
- availability_90
- availability_365
- calendar_last_scraped

**cost by neighborhood Fields to keep and clean**
- remove "2019" and "YTD" columns

**cost by bedrooms fields to keep and clean**
- Keep the # of Bedrooms and Median Price fields


### Rename Columns

**AirBnB listings**
- No renaming needed

**cost_by_neighborhood**
- Rename "2020" colums to use underscore and no caps

**cost by bedrooms**
- Rename the columns to be lowerase with underscores

### Filter Rows

**AirBnB listings**
- Only keep the rows where room_type = "Entire home/apt"
- Remove rows where bedrooms is blank

**cost by neighborhood**
- remove rows where neighborhood doesn't match AirBnB data

### Create Calculated Fields

**AirBnB listings**
- Create NOI field
- Create RPP field

### Combine datasets

Combine the cost by neighborhood and bost by bedrooms into one dataset